packages/jest-haste-map/src/ModuleMap.ts

```diff
@@ -3,53 +3,50 @@
  *
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
- *
- * @flow
  */
 
-import type {Path} from 'types/Config';
-import type {
+import {Config} from '@jest/types';
+import {
   DuplicatesSet,
   HTypeValue,
   ModuleMetaData,
   RawModuleMap,
   ModuleMapData,
   DuplicatesIndex,
   MockData,
-} from 'types/HasteMap';
+} from './types';
 
 import * as fastPath from './lib/fast_path';
 import H from './constants';
 
-const EMPTY_OBJ = {};
+const EMPTY_OBJ = {} as {[key: string]: any};
 const EMPTY_MAP = new Map();
 
-export opaque type SerializableModuleMap = {
+type ValueType<T> = T extends Map<T, infer V> ? V : T;
+
+export type SerializableModuleMap = {
   // There is no easier way to extract the type of the entries of a Map
-  duplicates: $Call<
-    typeof Array.from,
-    $Call<$PropertyType<DuplicatesIndex, 'entries'>>,
-  >,
-  map: $Call<typeof Array.from, $Call<$PropertyType<ModuleMapData, 'entries'>>>,
-  mocks: $Call<typeof Array.from, $Call<$PropertyType<MockData, 'entries'>>>,
-  rootDir: string,
+  duplicates: [string, ValueType<DuplicatesIndex>];
```

Comment:
duplicates has this signature: `Map<string, Map<string, Map<string, number>>>`. We'd need to call `Array.from` recursively if we wanted to properly `toJSON` and `fromJSON` it. Is this even working properly now? To me it looks like we loose all duplicates data on every serialization

Reply:
I agree it seems like this isn't working properly. Should probably investigate